src/sanity_check.rs

```diff
@@ -0,0 +1,24 @@
+use eyre::ensure;
+use url::Url;
+
+pub trait SanityCheck {
+    fn sanity_check(&self) -> eyre::Result<()>;
+}
+
+impl SanityCheck for Url {
```

Comment:
I've done it, but because we ask for the password before we have a change to construct the ArtifactoryConfig (and therefore before the sanity check runs), I split its initialization into two steps - first url+username, then ask for the password.